From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Javier Martinez Canillas <javierm@redhat.com>
Date: Thu, 11 Jul 2019 13:04:21 +0200
Subject: [PATCH] Revert "Disable multiboot, multiboot2, and linux16 modules on
 EFI builds."

This reverts commit 155d4e84604 which disabled building the multiboot and
multiboot2 modules on EFI builds. But that made the menu entries created
by the Xen package to stop working since they use the multiboot2 module.

The mentioned commit disabled building the multiboot{,2} modules because
they can be used to bypass the Secure Boot mechanism. But it's enough to
not include these modules in the grub2 EFI binary that's signed, which
is the case already in the grub2 package.

Having them as modules if the user installs the grub2-efi-x64-modules is
a valid use case. And since module loading isn't allowed when Secure Boot
is enabled, it doesn't represent any security threat.

Resolves: rhbz#1703872

Signed-off-by: Javier Martinez Canillas <javierm@redhat.com>
---
 grub-core/Makefile.core.def | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/grub-core/Makefile.core.def b/grub-core/Makefile.core.def
index 8a00c6177e1..b662312ca6f 100644
--- a/grub-core/Makefile.core.def
+++ b/grub-core/Makefile.core.def
@@ -1696,7 +1696,7 @@ module = {
 
   common = loader/multiboot.c;
   common = loader/multiboot_mbi2.c;
-  enable = i386_pc;
+  enable = x86;
   enable = mips;
 };
 
@@ -1705,7 +1705,7 @@ module = {
   common = loader/multiboot.c;
   x86 = loader/i386/multiboot_mbi.c;
   extra_dist = loader/multiboot_elfxx.c;
-  enable = i386_pc;
+  enable = x86;
 };
 
 module = {
